enhancements/agent-installer/host-based-mirror.md

```diff
@@ -0,0 +1,282 @@
+mirror. The bootstrap process will function in the usual manner with no
+external network required to run containers. No changes to pull secret or CA
+certificates will be needed to access this mirror.
+
+After the release image has been written to disk and bootstrap is complete, the
+container images must be made available upon reboot to complete the
+installation. It is envisioned that a method similar to the telco-ran-tools
+[ignition config override](https://github.com/openshift-kni/telco-ran-tools/blob/main/docs/ztp-precaching.md#43-nodesignitionconfigoverride)
+will be used. A script can be run using ignition config override to make the
+images available, create a registry in the cluster, and publish the images
+to the registry.
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+To create the tarfile of container images from within the openshift-installer
+the best choice will be `oc-mirror`. The binary can be extracted from
+the release image, it's well supported, and well documented. When
+`openshift-install agent create container-image-file` is run a basic
+imageset can be created from the release image. If additional operator or
+container images need to be included however, a configuration hook may need to
+be added to agent-config.yaml to define the imageset.  If the user runs
+`oc-mirror` directly without the `openshift-install` command then a
+user-defined imageset can be used and the resulting tarfile put into the
+assets directory.
+
+As noted above, the inclusion of the tarfile in ISO will be one of the
+challenges of this feature. It may be necessary to generate multiple ISOs
+when the tarfile is included. If so, the question will be as to how to
+mount these for the host. This section will be updated with more details
+as investigation continues.
+
+When the host boots, a service will run to make the container images
+available before other services attempt to pull these images. There
+are two potential paths, both with tradeoffs:
+1. The service creates a container to run a registry, either `docker-registry`
+or `quay` (note that only `docker-registry` is in the release image).
+`oc-mirror` is used to publish the contents of the tarfile to the registry.
+The `/etc/containers/registries.conf` is set up to use this
+local registry so all accesses to the container images use it.
+In this case, both the registry and `oc-mirror` binaries would need to be
+included in the ISO as they are not part of CoreOS.
+
+2. The service does not create a registry, instead it unpacks the
+tarfile and pushes the container images to container storage using
+`skopeo`. All pulls of the container images come directly from this
+container storage, aka [pre-pulled images](https://kubernetes.io/docs/concepts/containers/images/#pre-pulled-images).
+In this case the registry and `oc-mirror` binaries would not need to be
+included.
+
+For scenario 1, in a multi-node cluster configuration, e.g. 3 control
```

Comment:
what if node0 goes down before all the images were pulled from it? also user images